### ISSUE TITLE
Script error when start each channel

### ISSUE THREAD

**Issue by andrewhlk** [learningit/plugin.video.mediacorp#3]
From 1 April 2015, xinmsn is closed. Is this the cause of script error?

Regards,
andrew

**Comment by xobs** [comment]
Yes, the plugin needs to be rewritten to use Toggle.sg instead of xinmsn.

**Comment by learningit** [comment]
I've started looking at this. They have also moved to Silverlight which has its own challenges. I am able to play some of the videos from the toggle.sg site, but quite a few are DRM protected. I'm assessing if it's worth writing a new addon based on limited content - I probably will for viewing programs produced in Singapore, but the foreign content - including a lot of Channel8 will probably not play.

**Comment by dcapape** [comment]
Live TV of Toggle is working on Linux, then it isn't DRM and silverlight.
Just conecting proxy Channel 8 is working fine.

**Comment by learningit** [comment]
You might want to test again. Only some local programs are without DRM. Try something like Martha's Cooking School on channel 5. The returned videos are all .DVM which is a weird MS format- they'll play for 10-30 seconds then cut out as the DRM kicks in. Silverlight is not needed on any platform to play videos. A lot of both Channel 5 & 8 are non-local.

**Comment by learningit** [comment]
Try this url:
http://video.toggle.sg/en/series/jamie-s-15-minute-meals-s1/ep4/329859
I see what's returned as DRM. It plays about 20 secs before quitting.

**Comment by learningit** [comment]
This one plays fine as it is local:
http://video.toggle.sg/en/series/the-5-show-2015/ep26/330063

**Comment by learningit** [comment]
There are also low res .mp4s that don't require a proxy to play such as(Channel 8 example):
http://vod.toggletv.sg/c/tund/201502W-A/20150506_MONEY_WEEK_2015_EP4_ADD_MP4_40.mp4

The real issue I have is that the hi-res stuff on Channel 8 requires an internet proxy (as opposed to an addon level proxy) because the playable url is protected and the proxy needs to wrap the Kodi player. The addon will be quite complex to figure out what's local and then whether or not the m3u8 can be played and then finally resort to the low-res .mp4s. I never said an addon couldn't be done, I was questioning whether it was worth the effort.

**Comment by dcapape** [comment]
Yeah, i tested again now. First I test with Channel 8 live signal only, I didn't try VOD. 
It will be difficult to have an app like before with this issues. So sad.

**Comment by learningit** [comment]
You must have a much better proxy than I do. I'm able to open the live stream m3u8 files but I can't play the ts segments through a proxy. Anyway, I'll see what can be done, personally I just want local content and I think that can be done. A lot of channel 8's popular programs aren't local unfortunately.

**Comment by learningit** [comment]
Try this V1.4.1 and see if it works for you. The release notes are here:
http://forum.kodi.tv/showthread.php?tid=173754&pid=1999650#pid1999650

**Comment by learningit** [comment]
V1.4.3 has all channels catchup added. Please see:
http://forum.kodi.tv/showthread.php?tid=173754&pid=2000140#pid2000140
